lib/mobile_app_backend/route_branching/segment.ex

```diff
@@ -574,9 +574,9 @@ defmodule MobileAppBackend.RouteBranching.Segment do
   defp assign_segments([s], [0, 1], _), do: %{s => 0}
   defp assign_segments([s], [1, 2], _), do: %{s => 2}
   defp assign_segments([s], [0, 2], _), do: %{s => 0}
-  defp assign_segments([s1, s2], [t1, t2], _), do: %{s1 => t1, s2 => t2}
-  defp assign_segments([s1, s2], [0, 1, 2], _), do: %{s1 => 0, s2 => 2}
-  defp assign_segments([s1, s2, s3], [0, 1, 2], _), do: %{s1 => 0, s2 => 1, s3 => 2}
+  defp assign_segments([s1, s2], [t1, t2], _), do: %{s1 => t2, s2 => t1}
+  defp assign_segments([s1, s2], [0, 1, 2], _), do: %{s1 => 2, s2 => 0}
+  defp assign_segments([s1, s2, s3], [0, 1, 2], _), do: %{s1 => 2, s2 => 1, s3 => 0}
```

Comment:
should the single segment [0,2] case be flipped as well?
Did the 3 segment case need to be flipped? I think so for since hingham/hull looks good this way

Reply:
Maybe should just sort the `unconstrained_segments` being passed to `assign_segments`

Reply:
May as well flip the `[s], [0, 2]`, yeah. The 3 segment case is rare, and if Hingham/Hull has it and looks better flipped, then flipping it is correct.